<launch>
  <!-- Read BATTERY_MODEL from environment (default = MICRONIX) -->
  <arg name="battery_model" default="$(optenv BATTERY_MODEL ATC)" />

  <!-- Common topic name to publish battery state -->
  <arg name="battery_topic" default="/battery_state" />

  <!-- Launch Micronix BMS node -->
  <group if="$(eval arg('battery_model') == 'MICRONIX')">
    <node pkg="bms_interface"
          type="micronix_bms_node"
          name="micronix_bms_node"
          output="screen">
      <remap from="/battery_state" to="$(arg battery_topic)" />
    </node>
  </group>

  <!-- Launch ATC BMS node -->
  <group if="$(eval arg('battery_model') == 'ATC')">
    <node pkg="bms_interface"
          type="atc_bms_node"
          name="atc_bms_node"
          output="screen">
      <remap from="/battery_state" to="$(arg battery_topic)" />
    </node>
  </group>

  <!-- Launch APERION BMS CAN parser node -->
  <group if="$(eval arg('battery_model') == 'APERION')">
    <node pkg="bms_interface"
          type="aperion_bms_node"
          name="aperion_bms_node"
          output="screen">
      <remap from="/battery_state" to="$(arg battery_topic)" />
    </node>
  </group>

<group if="$(eval arg('battery_model') == 'MEL1')">
  <node pkg="bms_interface"
        type="mel_bms_node"
        name="mel_bms_node"
        output="screen">
    <remap from="/battery_state" to="$(arg battery_topic)" />
  </node>
</group>

</launch>
